### PR TITLE
[Merged by Bors] - refactor: generalize Abs lemmas from rings to groups

### PR DESCRIPTION
**Pull request by alreadydone**
Four lemmas are moved from [Algebra/Order/Monoid/Defs.lean](https://github.com/leanprover-community/mathlib4/pull/7976/files#diff-b7e2a9cad39a929d367ebafb50a8fee9a683847bfe6cf4965d26c8b73d938162) to [Algebra/Order/Group/Defs.lean](https://github.com/leanprover-community/mathlib4/pull/7976/files#diff-1ea79e1c68d33cde0e0d03b5cd88acada0fe81a37be3c91e0dbdf5734e81b824) and generalized

Four lemmas are moved from [Algebra/Order/Ring/Abs.lean](https://github.com/leanprover-community/mathlib4/pull/7976/files#diff-df3e020492662385d4cf526fc26cd5d5292e333cecb7aa35591abbf597ff36d7) to [Algebra/Order/Group/Abs.lean](https://github.com/leanprover-community/mathlib4/pull/7976/files#diff-04fe7d4ab1d9e83e80baaa4ff56278e736e3ec0a4ddace3fa767ddc86c69821b) and generalized

Four lemmas are added in [Algebra/Order/Monoid/Defs.lean](https://github.com/leanprover-community/mathlib4/pull/7976/files#diff-b7e2a9cad39a929d367ebafb50a8fee9a683847bfe6cf4965d26c8b73d938162). They're special cases of [one_le_pow_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GroupPower/Order.html#one_le_pow_iff), but I can't import the file without offending assert_not_exists.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alreadydone** [comment]
@YaelDillies said on Zulip:
> Anyway I have a full rewrite of the files in question coming up, so don't worry too much about it.

Does it include what's changed in this PR?

I know that lemmas here could probably be generalized to CovariantClass without using the bundled algebra+order classes, but I don't have time to do that.

File: Mathlib/Algebra/Order/Group/Abs.lean [hunk 1]
```diff
@@ -368,4 +369,32 @@ theorem eq_of_abs_sub_nonpos (h : |a - b| ≤ 0) : a = b :=
...
+  by_cases h : 0 ≤ a
+  · left
+    exact ⟨abs_eq_self.mpr h, h⟩
+  · right
+    push_neg at h
+    exact ⟨abs_eq_neg_self.mpr (le_of_lt h), h⟩
+#align abs_cases abs_cases
+
+@[simp]
+theorem max_zero_add_max_neg_zero_eq_abs_self (a : α) : max a 0 + max (-a) 0 = |a| := by
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Group/Abs.lean:395]: This lemma already exists. See `posPart` and `negPart`.

**Review by YaelDillies — COMMENTED** [review]
Yes, it includes what's in that PR but feel free to merge this first if you need it soon since I don't currently have much time.

File: Mathlib/Algebra/Order/Group/Abs.lean (hunk 1 above)
Comment (alreadydone) [reply Mathlib/Algebra/Order/Group/Abs.lean:395]: This lemma is moved, not added. Loogle only finds [four declarations](https://loogle.lean-lang.org/?q=PosPart) about PosPart. Do you mean a version of this lemma that works for AddCommGroup already exist elsewhere?

**Comment by alreadydone** [comment]
!bench

File: Mathlib/Algebra/Order/Group/Abs.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/Group/Abs.lean:394]: Your lemma is defeq to (the `symm` of) `LatticeOrderedGroup.pos_add_neg`.

Comment (alreadydone) [reply Mathlib/Algebra/Order/Group/Abs.lean:394]: I see, thanks! Should have searched https://loogle.lean-lang.org/?q=PosPart.pos

Comment (alreadydone) [reply Mathlib/Algebra/Order/Group/Abs.lean:394]: Unfortunately, Mathlib.Algebra.Order.LatticeGroup imports this file. The lemma is used at [two occasions](https://github.com/search?q=repo%3Aleanprover-community%2Fmathlib4%20max_zero_add_max_neg_zero_eq_abs_self&type=code) and aligned so I'm inclined to not move it for now.

**Comment by PatrickMassot** [comment]
bors merge

**Pull request closed without merging**